src/matches.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+export type NormalizerFn = (textToNormalize: string) => string;
+
+function matches(
+  textToMatch: string,
+  matcher: string | RegExp,
+  normalizer: NormalizerFn,
+  exact?: boolean = true
+) {
+  if (typeof textToMatch !== 'string') {
+    return false;
+  }
+
+  const normalizedText = normalizer(textToMatch);
+  if (typeof matcher === 'string') {
+    return exact
+      ? normalizedText === matcher
+      : normalizedText.toLowerCase().includes(matcher.toLowerCase());
+  } else {
+    return matcher.test(normalizedText);
+  }
+}
+
+function getDefaultNormalizer({
+  trim = true,
+  collapseWhitespace = true,
+}: NormalizerInput = {}): NormalizerFn {
+  return (text: string) => {
+    let normalizedText = text;
+    normalizedText = trim ? normalizedText.trim() : normalizedText;
+    normalizedText = collapseWhitespace
+      ? normalizedText.replace(/\s+/g, ' ')
+      : normalizedText;
+    return normalizedText;
+  };
+}
+
+type NormalizerInput = {
+  trim?: boolean,
+  collapseWhitespace?: boolean,
+  normalizer?: NormalizerFn,
+};
+
+function makeNormalizer({
+  trim,
+  collapseWhitespace,
+  normalizer,
+}: NormalizerInput = {}): NormalizerFn {
+  if (normalizer) {
+    if (
+      typeof trim !== 'undefined' ||
+      typeof collapseWhitespace !== 'undefined'
```

Comment:
It's cheaper to not typeof 
```suggestion
      trim !== undefined || collapseWhitespace !== undefined
```